Tests/Clappr_Tests/AVFoundationPlaybackTests.swift

```diff
@@ -933,6 +933,207 @@ class AVFoundationPlaybackTests: QuickSpec {
                 }
             }
             
+            describe("#AVFoundationPlayback states") {
+                describe("#idle") {
+                    context("when ready to play") {
+                        it("current state must be idle") {
+                            let playback = AVFoundationPlayback()
+                            
+                            expect(playback.currentState).to(equal(.idle))
+                        }
+                    }
+
+                    context("when play is called") {
+                        it("changes isPlaying to true") {
+                            let options = [kSourceUrl: "http://localhost:8080/sample.m3u8"]
+                            let playback = AVFoundationPlayback(options: options)
+
+                            playback.play()
+
+                            expect(playback.isPlaying).to(beTrue())
```

Comment:
Shouldn't we also test what is the value of `isPaused` and `isBuffering` for each case? In that way, we can ensure that every representation of the state is ok.

What do you guys think?

Reply:
Hey @leandroalonso ,
You are right, we should test those values to make sure about the state representation.
But when we did that, we found that in the moment of the test those properties may have a delay to change their value.

What we did this time, we added a timeout for the expectation to make it possible to test.
Take a look into the last commit and let me know if it's how you expect.